src/components/result-ui/summary.js

```diff
@@ -0,0 +1,82 @@
+  const flattened = {};
+  Object.entries(obj).forEach(([property, value]) => {
+    const key = `${previous.length ? `${previous}.` : ''}${property}`;
+    let assign;
+    if (value && typeof value === 'object') {
+      assign = flattenObject(value, key);
+    }  else {
+      assign = {[key]: value};
+    }
+    Object.assign(flattened, assign);
+  });
+  return flattened;
+};
+
+const FieldRow = ({ fieldPath, fieldValue, focused, focus, defocus }) => {
+  return (
+    <>
+      <div className={"result-row-path"}>
+        {focused
+          ? <a className={"result-row-field-focus-link"} onClick={focus} title="Hide other fields"><span className={"arrow"}>&uArr;</span>{fieldPath}</a>
+          : <a className={"result-row-field-focus-link"} onClick={defocus} title="Show all fields"><span className={"arrow"}>&dArr;</span>{fieldPath}</a>
+        }
+      </div>
+      <div className={"result-row-value"}>{JSON.stringify(fieldValue)}</div>
+    </>
+  );
+};
+
+const Summary = ({data, focusPath, defaultZoom}) => {
+  const [zoom, setZoom] = useState(defaultZoom);
+  const [focus, setFocus] = useState(focusPath);
+  const rows = [data]
+    .flat()
+    .filter(obj => !zoom || obj.matches(zoom));
+  return (rows.length && <div className={"results"}>
+    <ul>
+      {rows.map((resourceObject, i) => {
+        const type = resourceObject.getType(), id = resourceObject.getID();
+        const attributes = Object.entries(flattenObject(resourceObject.getAttributes())).filter(([fieldPath]) => !focus || fieldPath === focus);
+        const links = Object.entries(resourceObject.getOutgoingLinks());
+        return (
+          <li key={`result-row-${i}`} className={"result-row"} title={`${id} (${type})`}>
+            <ul>
+            {attributes.map(([fieldPath, fieldValue], j) => (
+              <li key={`result-row-${i}-field-${j}`} className={"result-row-field"}>
+                <FieldRow
+                  fieldPath={fieldPath}
+                  fieldValue={fieldValue}
+                  focused={!focus || fieldPath !== focus}
+                  focus={() => {
+                    setFocus(fieldPath);
+                    setZoom(null);
+                  }}
+                  defocus={() => {
+                    setFocus(null);
+                  }}
+                />
+              </li>
+            ))}
+            </ul>
+            <div className={"result-row-actions"}>
+              <ul>
+                {(!zoom || !resourceObject.matches(zoom)) && <li key={'zoom'}><button onClick={() => { setZoom({type, id}); setFocus(null) }}>Expand result</button></li>}
+                {links.map(([key, link], index) => (
+                  <li key={`link-${index}`}>
+                    <LinkElement link={link} />
+                  </li>
+                ))}
+              </ul></div>
+          </li>
+        )
+      })}
+    </ul>
+    {zoom && <button onClick={() => setZoom(null)}>Show all results</button>}
```

Comment:
This should maybe be at the top instead, or sticky somehow so that you can see it. In practice when one result is expanded you might miss it at the very bottom.

Reply:
Heh, I actually had this at the top first and it felt weird because you'd scroll to the bottom, it'd go out of view and there wasn't a way to really know that you could "load more". So I put it at the bottom for that case.

I didn't think about making it sticky though. That's just the kind of thing I could use your help with :D